frontend_app_file/src/container/File.jsx

```diff
@@ -778,111 +778,119 @@ class File extends React.Component {
+                  creationDateFormatted={(new Date(state.content.created)).toLocaleString(props.i18n.language)}
+                  lastModification={displayDistanceDate(state.content.modified, state.loggedUser.lang)}
+                  lastModificationFormatted={(new Date(state.content.modified)).toLocaleString(props.i18n.language)}
+                  description={state.content.raw_content}
+                  displayChangeDescriptionBtn={state.loggedUser.userRoleIdInWorkspace >= 2}
+                  disableChangeDescription={!state.content.is_editable}
+                  onClickValidateNewDescription={this.handleClickValidateNewDescription}
+                  key={'FileProperties'}
+                />
+              )
+            }]
     } else {
-      return [
-        {
-          id: 'timeline',
-          label: props.t('Timeline'),
-          icon: 'fa-history',
-          children: <Timeline
-            customClass={`${state.config.slug}__contentpage`}
-            customColor={state.config.hexcolor}
-            loggedUser={state.loggedUser}
-            timelineData={state.timeline}
-            newComment={state.newComment}
-            disableComment={state.mode === MODE.REVISION || state.mode === MODE.EDIT || !state.content.is_editable}
-            availableStatusList={state.config.availableStatuses}
-            wysiwyg={state.timelineWysiwyg}
-            onChangeNewComment={this.handleChangeNewComment}
-            onClickValidateNewCommentBtn={this.handleClickValidateNewCommentBtn}
-            onClickWysiwygBtn={this.handleToggleWysiwyg}
-            onClickRevisionBtn={this.handleClickShowRevision}
-            shouldScrollToBottom={state.mode !== MODE.REVISION}
-          />
-        },
-        {
-          id: 'properties',
-          label: props.t('Properties'),
-          icon: 'fa-info-circle',
-          children: <FileProperties
-            color={state.config.hexcolor}
-            fileType={state.content.file_extension}
-            fileSize={displayFileSize(state.content.size)}
-            filePageNb={state.content.page_nb}
-            activesShares={state.content.actives_shares}
-            creationDateFormattedWithTime={(new Date(state.content.created)).toLocaleString(props.i18n.language, { day: '2-digit', month: '2-digit', year: 'numeric' })}
-            creationDateFormatted={(new Date(state.content.created)).toLocaleString(props.i18n.language)}
-            lastModification={displayDistanceDate(state.content.modified, state.loggedUser.lang)}
-            lastModificationFormatted={(new Date(state.content.modified)).toLocaleString(props.i18n.language)}
-            description={state.content.raw_content}
-            displayChangeDescriptionBtn={state.loggedUser.userRoleIdInWorkspace >= 2}
-            disableChangeDescription={!state.content.is_editable}
-            onClickValidateNewDescription={this.handleClickValidateNewDescription}
-          />
-        }
-      ]
+      return [{
```

Comment:
in your "if" and your "else" there are 2 components that are the same (Timeline and FileProperty).

I haven't check if they have exactly the same props but if they have, put each array elements in a variable and only return an array of these variables. It'll save a LOT of lines of code.